### DISCUSSION TITLE
Task stalled on "no status" keeps the DAG run status as "running"

### DISCUSSION THREAD

**Discussion by alejandrofm** [apache/airflow#30492] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

Hi, I'm in Airflow 2.4.2, couldn't find a similar issue, when a DAG is configured with "depends_on_past = True", and the task is blocked on status "no status" the DAG run stays in status=Running, using a running slot.
<img width="108" alt="image" src="https://user-images.githubusercontent.com/19591174/230212619-6645d8af-4bd5-44c2-9038-82d1d72e5280.png">

### What you think should happen instead

The task+DAG run should be on a status like "stalled" not using a run slot and the dag shouldn't be on status running.

### How to reproduce

Set a DAG with depends_on_past =  True
Make a task fail so he next cant start (status=no status)
the DAG run will be "running" infinitelly

### Operating System

apache/airflow:2.4.2-python3.10

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by hussein-awala** [comment]
First of all, this is not a bug, rather it is a new feature request.

As long as the dag run has at least one unfinished task(state not in [success, failed, skipped, upstream_failed, removed]), and the dag run is not timed out and not manually cleared , its state should remain running.

For the new proposed state `stalled`, I think this can add a lot of complexity to the scheduler job, which should process the dag runs with the state `stalled` in the critical section like the state `running` to check if the tasks are unblocked or they still somehow blocked, and should not run any task in the stalled dag runs before passing them to running, respecting that the count of running tasks is less than or equal to the max active dag runs.

IMHO this can be a good feature in some use cases, especially in large dags that have a small number of tasks with depend_on_past or with a small pool size. I convert it to discussion.

**Reply by potiuk** [reply]
Yes. It could be nice feature, but it has a bit more detailed description and proposal how to implement it. Because it willl need some scheduler processing to make it works at scale. Likely it will require some  changes to extra work done by scheduler. @alejandrofm - I propose, if you woul like to pursue it, to bring that discussion to devlist, as this discussion is also about changing the base state machine of Airlfow tasks:  https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/tasks.html#task-instances

**Comment by alejandrofm** [comment]
I'm diving into some deadlocks I found, this is similar to this: https://github.com/apache/airflow/issues/29524
I don't know the difficulty of implementing this, but I will be a good addition.
Thanks!

**Comment by awadhoot-sq** [comment]
what is an alternative way to implement this requirement? i want DAG to stop running if there are stalled tasks, may be depends_on_past is not the right option for me, any suggestions?

**Reply by awadhoot-sq** [reply]
@alejandrofm did you find any solution to implement your requirment?

**Reply by alejandrofm** [reply]
I'm still on 2.4.2 don't know if there was any advance, so I will update next week to 2.7.1. No solution on my side so far

**Reply by awadhoot-sq** [reply]
I am exploring Airflow REST API for this, and set status programmatically
